utils/rigid_origami_utils.py

```diff
@@ -0,0 +1,513 @@
+        cr_edges = [[crease_lines.edges[j] for j in crease_indices[i]] \
+                    for i in range(len(crease_indices))]
+        
+        # Create theta (between edges) and rho (edge angle) list
+        theta_list = [cls.__calc_theta_angles(obj.verts, cr_edges[i], idx) \
+                        for i, idx in enumerate(indices)]
+        rho_list = [[crease_lines.angles[j] for j in crease_indices[i]] \
+                        for i in range(len(indices))]
+        drho_list = [[crease_lines.angles[j]+np.pi*0.5 for j in crease_indices[i]] \
+                        for i in range(len(indices))]
+
+        # Create list of InsideVertex class
+        inside_verts = [InsideVertex(obj.verts[idx], idx, cr_edges[i], \
+                        crease_indices[i], theta_list[i], rho_list[i], \
+                        drho_list[i]) \
+                        for i, idx in enumerate(indices)]
+
+        return inside_verts
+    
+    # Function to get crease lines around the vertex
+    # https://blender.stackexchange.com/questions/92406/circular-order-of-edges-around-vertex
+    @classmethod
+    def __GetCreaseLinesAroundVertex(cls, crease_lines, inside_vert_indices, obj):
+
+        # Sort link_edges around each vertices in counter-clockwise order
+        bm_edges_ccws = []
+        for i in inside_vert_indices:
+            bm_vert = obj.bm.verts[i]
+            bm_vert.link_edges.index_update()
+            
+            edges_counterclockwise_order = []
+            bm_edge = bm_vert.link_edges[0]
+            while bm_edge not in edges_counterclockwise_order:
+                edges_counterclockwise_order.append(bm_edge)
+                bm_edge = cls.__GetRightSideEdgeAroundVertex(bm_edge, bm_vert, obj)
+            bm_edges_ccws.append(edges_counterclockwise_order)
+            
+        crease_edge_indices = [[] for i in inside_vert_indices]
+        outer_verts_indices = [idx for idx, vert in enumerate(obj.bm.verts) if vert.is_boundary]
+        target_indices = [idx for idx in inside_vert_indices]
+        
+        # Rotate the sorted edge indices to enable the paper foldable
+        while len(target_indices) > 0:
+
+            target_list_idx = [inside_vert_indices.index(target_idx) for target_idx in target_indices]
+            target_ccws = [ccw for i, ccw in enumerate(bm_edges_ccws) if i in target_list_idx]
+            
+            # Select outermost indices not yet to be processed
+            indices_outermost = [idx for idx, (target_ccw, v_idx) in \
+                                        enumerate(zip(target_ccws, target_indices)) \
+                                        if any([bme.other_vert(obj.bm.verts[v_idx]).index \
+                                        in outer_verts_indices for bme in target_ccw])]
+                                        
+            # Roll the edge order around each vertices
+            for idx, target_ccw in enumerate(target_ccws):
+                if not idx in indices_outermost:
```

Comment:
```suggestion
                if idx not in indices_outermost:
```
A bit better to read

Reply:
Thank you :). I should have learned more about python before this ...